ouroboros-network/src/Ouroboros/Network/NodeToClient.hs

```diff
@@ -386,3 +389,31 @@ ncSubscriptionWorker_V1
           versionData
           (DictVersion nodeToClientCodecCBORTerm)
           application)
+
+-- | 'ErrorPolicies' for client application.  Additional rules can be added by
+-- means of a 'Semigroup' instance of 'ErrorPolicies'.
+--
+-- This is a good set of policies only if the client tries to connect to
+-- a single node, if it would be configured to connect to multiple nodes one
+-- should suspend the consumer instead of killing the whole application.
+--
+networkErrorPolicies :: ErrorPolicies Socket.SockAddr a
+networkErrorPolicies = ErrorPolicies
+    { epAppErrorPolicies = [
+        -- Handshake client protocol error: we either did not recognise received
+        -- version or we refused it.  This is only for outbound connections to
+        -- a local node, thus we throw the exception.
+        ErrorPolicy
+          $ \(_ :: HandshakeClientProtocolError NodeToClientVersion)
+                -> Just ourBug
+      ]
+    , epConErrorPolicies = [
+        -- If an 'IOException' is thrown by the 'connect' call, kill the
+        -- application.
+        ErrorPolicy $ \(_ :: IOException) -> Just ourBug
```

Comment:
Does this mean that a temporary connection error experienced by `ncSubscriptionWorker` will cause to node to restart?
I would expect this to be a short timeout.

Reply:
This will kill the node if the `connect` throws an exception. So yes, if there was a network failure, and the node tries to connect and the failure still persists it will halt the node.

Also I am not handling any `IOExceptions` in the `epAppErrorPolicies` and we should, for the reason you mentioned.

Reply:
I will modify the policy, and justify it in a haddock.